examples/ex9p.cpp

```diff
@@ -16,6 +16,8 @@
 //    mpirun -np 4 ex9p -m ../data/disc-nurbs.mesh -p 2 -rp 1 -dt 0.005 -tf 9
 //    mpirun -np 4 ex9p -m ../data/periodic-square.mesh -p 3 -rp 2 -dt 0.0025 -tf 9 -vs 20
 //    mpirun -np 4 ex9p -m ../data/periodic-cube.mesh -p 0 -o 2 -rp 1 -dt 0.01 -tf 8
+//    mpirun -np 4 ex9p -m ../data/periodic-square.msh -p 0 -rs 2 -dt 0.005 -tf 2
+//    mpirun -np 4 ex9p -m ../data/periodic-cube.msh -p 0 -rs 1 -o 2 -tf 2 -no-vis -paraview
```

Comment:
Another instance...

```suggestion
//    mpirun -np 4 ex9p -m ../data/periodic-cube.msh -p 0 -rs 1 -o 2 -tf 2 -paraview
```

Reply:
I'll take a look at it.

Reply:
I think the trick is to add boundary elements on both periodic boundaries with opposite orientations. I don't know if we can convince Gmsh to do this but we can probably add one surface or the other which would be better than nothing.

Reply:
OK, Gmsh will be happy to create the missing surfaces. Unfortunately, it's more complicated than that. Our Gmsh reader automatically removes internal surface from periodic meshes. This was done because the other periodic meshes in the `data` directory do not contain boundary elements.

We could switch this behavior in the Gmsh reader but this would break some of our existing sample runs involving periodic Gmsh meshes in `ex1` and `ex11p` (and perhaps more) because these examples apply homogeneous Dirichlet BCs on any boundary surfaces they find in the mesh which would include these periodic surfaces.

We might be able to control the Gmsh reader behavior with respect to internal boundaries but I'm not sure how we would pass this option to it through the `Mesh` constructor.

Reply:
A longer term solution might involve decoupling of boundary elements for discretization and "fake boundary elements" used for visualization of internal or periodic surfaces. These two things seem pretty separate to me, and it probably makes sense to clearly distinguish between them in the code.

Reply:
I think issue #2231 is also related to "fake boundary elements".

Reply:
Is it possible to solve this alternatively in GLVis?

Reply:
I think it would be possible but perhaps expensive. GLVis might need to visualize volume elements when boundary elements are not present. This might involve determining which elements are hidden by others but this would of course change as the mesh rotates.